packages/smooth_app/lib/pages/product/simple_input_text_field.dart

```diff
@@ -33,9 +33,7 @@ class SimpleInputTextField extends StatelessWidget {
           mainAxisSize: MainAxisSize.max,
           children: <Widget>[
             SizedBox(
-              width: constraints.maxWidth -
-                  LARGE_SPACE -
-                  (withClearButton ? MINIMUM_TOUCH_SIZE : 0),
+              width: constraints.maxWidth - LARGE_SPACE - MINIMUM_TOUCH_SIZE,
```

Comment:
Same thing here: use `Expanded` / `flex` for the "as big as possible" item, and a fixed size for the trailing widget.